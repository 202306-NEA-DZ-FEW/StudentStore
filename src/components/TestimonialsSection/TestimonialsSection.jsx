import React, { useState, useEffect } from "react";
import {
    IoIosArrowDroprightCircle,
    IoIosArrowDropleftCircle,
} from "react-icons/io";
import { FaQuoteRight } from "react-icons/fa";
import data from "@/util/data";
import Image from "next/image";
function TestimonialsSection() {
    const [people, setPeople] = useState(data);
    const [index, setIndex] = useState(0);

    useEffect(() => {
        const lastIndex = people.length - 1;
        if (index < 0) {
            setIndex(lastIndex);
        }
        if (index > lastIndex) {
            setIndex(0);
        }
    }, [index, people]);

    useEffect(() => {
        let slider = setInterval(() => {
            setIndex(index + 1);
        }, 5000);
        return () => {
            clearInterval(slider);
        };
    }, [index]);

    return (
        <section className='py-6 mx-auto'>
            <div className='mx-auto mt-0 w-[80vw] lg:w-[65vw] h-[350px] lg:[300px]max-w-[800px] text-center relative flex items-center overflow-hidden'>
                {people.map((person, personIndex) => {
                    const { id, image, name, quote } = person;

                    let position = "translate-x-full";
                    if (personIndex === index) {
                        position = "translate-x-0";
                    }
                    if (
                        personIndex === index - 1 ||
                        (index === 0 && personIndex === people.length - 1)
                    ) {
                        position = "-translate-x-full";
                    }

                    return (
                        <article
                            className={`${position} absolute h-auto w-full py-3 rounded-lg bg-white`}
                            key={id}
                        >
                            <Image
                                src={image}
                                alt={name}
                                width={100}
                                height={100}
                                className='object-cover h-16 w-16 mx-auto block border-2 border-[#7874F2] rounded-full my-4'
                            />

                            <h4 className='uppercase mb-[0.25rem] text-[1rem] font-bold text-[#7874F2]'>
                                {name}
                            </h4>
                            <p className='max-w-[15rem] sm:max-w-[25rem] md:max-w-[30rem] lg:max-w-[35rem] xl:max-w-[40rem] mx-auto  leading-loose'>
                                {quote}
                            </p>
                            <div className='flex justify-center text-[#7874F2]'>
                                <FaQuoteRight className='text-3xl mt-2' />
                            </div>
                        </article>
                    );
                })}
                <button
                    className='absolute top-[50%] -translate-y-1/2 w-[1.25rem] h-[1.25rem] grid place-items-center text-[1rem] cursor-pointer text-gray-400 hover:text-[#7874F2] left-0'
                    onClick={() => setIndex(index - 1)}
                >
                    <IoIosArrowDropleftCircle className='text-3xl' />
                </button>
                <button
                    className='absolute top-[50%] -translate-y-1/2 w-[1.25rem] h-[1.25rem] grid place-items-center text-[1rem] cursor-pointer text-gray-400 hover:text-[#7874F2] right-3'
                    onClick={() => setIndex(index + 1)}
                >
                    <IoIosArrowDroprightCircle className='text-3xl' />
                </button>
            </div>
        </section>
    );
}

export default TestimonialsSection;
