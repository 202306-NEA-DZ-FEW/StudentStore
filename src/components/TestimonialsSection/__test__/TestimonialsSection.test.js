import renderer from "react-test-renderer";
import TestimonialsSection from "../TestimonialsSection";

it("renders correctly", () => {
    const tree = renderer.create(<TestimonialsSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
