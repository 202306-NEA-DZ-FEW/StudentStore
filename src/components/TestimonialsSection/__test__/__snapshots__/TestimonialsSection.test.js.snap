// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="py-6 mx-auto"
>
  <div
    className="mx-auto mt-0 w-[80vw] lg:w-[65vw] h-[350px] lg:[300px]max-w-[800px] text-center relative flex items-center overflow-hidden"
  >
    <article
      className="translate-x-0 absolute h-auto w-full py-3 rounded-lg bg-white"
    >
      <img
        alt="Mazin Ismail"
        className="object-cover h-16 w-16 mx-auto block border-2 border-[#7874F2] rounded-full my-4"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Ftestimonial1.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Ftestimonial1.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Ftestimonial1.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="uppercase mb-[0.25rem] text-[1rem] font-bold text-[#7874F2]"
      >
        Mazin Ismail
      </h4>
      <p
        className="max-w-[15rem] sm:max-w-[25rem] md:max-w-[30rem] lg:max-w-[35rem] xl:max-w-[40rem] mx-auto  leading-loose"
      >
        Affordable books, gadgets, and a welcoming student community. This marketplace makes student life easier and more budget-friendly. Highly recommend!
      </p>
      <div
        className="flex justify-center text-[#7874F2]"
      >
        <svg
          className="text-3xl mt-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H336c-26.5 0-48 21.5-48 48v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48zm-288 0H48C21.5 32 0 53.5 0 80v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </article>
    <article
      className="translate-x-full absolute h-auto w-full py-3 rounded-lg bg-white"
    >
      <img
        alt="Hoda Khaliq"
        className="object-cover h-16 w-16 mx-auto block border-2 border-[#7874F2] rounded-full my-4"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Ftestimonial2.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Ftestimonial2.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Ftestimonial2.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="uppercase mb-[0.25rem] text-[1rem] font-bold text-[#7874F2]"
      >
        Hoda Khaliq
      </h4>
      <p
        className="max-w-[15rem] sm:max-w-[25rem] md:max-w-[30rem] lg:max-w-[35rem] xl:max-w-[40rem] mx-auto  leading-loose"
      >
        This platform is a lifesaver for broke college students like me. I sold old stuff and bought what I needed at affordable prices. Highly recommended!
      </p>
      <div
        className="flex justify-center text-[#7874F2]"
      >
        <svg
          className="text-3xl mt-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H336c-26.5 0-48 21.5-48 48v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48zm-288 0H48C21.5 32 0 53.5 0 80v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </article>
    <article
      className="translate-x-full absolute h-auto w-full py-3 rounded-lg bg-white"
    >
      <img
        alt="Khalil Yousef"
        className="object-cover h-16 w-16 mx-auto block border-2 border-[#7874F2] rounded-full my-4"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Ftestimonial3.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Ftestimonial3.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Ftestimonial3.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="uppercase mb-[0.25rem] text-[1rem] font-bold text-[#7874F2]"
      >
        Khalil Yousef
      </h4>
      <p
        className="max-w-[15rem] sm:max-w-[25rem] md:max-w-[30rem] lg:max-w-[35rem] xl:max-w-[40rem] mx-auto  leading-loose"
      >
        As a busy student, I appreciate the convenience of this website. I borrowed games and textbooks without a hitch. It's a game-changer for student life!
      </p>
      <div
        className="flex justify-center text-[#7874F2]"
      >
        <svg
          className="text-3xl mt-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H336c-26.5 0-48 21.5-48 48v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48zm-288 0H48C21.5 32 0 53.5 0 80v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </article>
    <article
      className="-translate-x-full absolute h-auto w-full py-3 rounded-lg bg-white"
    >
      <img
        alt="Radiya Benmoghaddam"
        className="object-cover h-16 w-16 mx-auto block border-2 border-[#7874F2] rounded-full my-4"
        data-nimg="1"
        decoding="async"
        height={100}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Ftestimonial4.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimages%2Ftestimonial4.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Ftestimonial4.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={100}
      />
      <h4
        className="uppercase mb-[0.25rem] text-[1rem] font-bold text-[#7874F2]"
      >
        Radiya Benmoghaddam
      </h4>
      <p
        className="max-w-[15rem] sm:max-w-[25rem] md:max-w-[30rem] lg:max-w-[35rem] xl:max-w-[40rem] mx-auto  leading-loose"
      >
        I've been amazed by the incredible support from this website's community. I not only found what I needed but also made friends along the way. It's a fantastic platform for students!
      </p>
      <div
        className="flex justify-center text-[#7874F2]"
      >
        <svg
          className="text-3xl mt-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H336c-26.5 0-48 21.5-48 48v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48zm-288 0H48C21.5 32 0 53.5 0 80v128c0 26.5 21.5 48 48 48h80v64c0 35.3-28.7 64-64 64h-8c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24h8c88.4 0 160-71.6 160-160V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </div>
    </article>
    <button
      className="absolute top-[50%] -translate-y-1/2 w-[1.25rem] h-[1.25rem] grid place-items-center text-[1rem] cursor-pointer text-gray-400 hover:text-[#7874F2] left-0"
      onClick={[Function]}
    >
      <svg
        className="text-3xl"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm43.4 289.1c7.5 7.5 7.5 19.8 0 27.3-3.8 3.8-8.7 5.6-13.6 5.6s-9.9-1.9-13.7-5.7l-94-94.3c-6.9-7.6-6.7-19.3.6-26.6l95.4-95.7c7.5-7.5 19.7-7.6 27.3 0 7.5 7.5 7.6 19.7 0 27.3l-81.9 81 79.9 81.1z"
        />
      </svg>
    </button>
    <button
      className="absolute top-[50%] -translate-y-1/2 w-[1.25rem] h-[1.25rem] grid place-items-center text-[1rem] cursor-pointer text-gray-400 hover:text-[#7874F2] right-3"
      onClick={[Function]}
    >
      <svg
        className="text-3xl"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm244.5 0l-81.9-81.1c-7.5-7.5-7.5-19.8 0-27.3s19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81z"
        />
      </svg>
    </button>
  </div>
</section>
`;
